modules/git/Makefile

```diff
@@ -4,4 +4,9 @@ GIT:= $(shell which git)
 ## Update submodules
 git\:submodules-update:
 	$(call assert-set,GIT)
-	$(GIT) submodule update --init --remote
\ No newline at end of file
+	$(GIT) submodule update --init --remote
+
+.PHONY: git\:aliases-update
+## Update git aliases
+git\:aliases-update:
+	@$(GIT) config --global --add alias.permission-reset '!git diff -p -R --no-color | grep -E "^(diff|(old|new) mode)" --color=never | git apply'
```

Comment:
This fix does not work on travis, but does work locally for me. With this, you can just run `git permission-reset`
see: http://stackoverflow.com/questions/2517339/how-to-recover-the-file-permissions-to-what-git-thinks-the-file-should-be